### PR TITLE
Migrate to eslint

### PR DESCRIPTION
**Pull request by raajnadar**
Reason for PR

https://github.com/palantir/tslint/issues/4534

What did I do?

- Added eslint and prettier config.
- Fixed lint issues.
- Added npm script to run lint test with CI.

While setting up I found few typescript warning which we need to fix.

### CODE REVIEW

**Comment by raajnadar** [comment]
@YIZHUANG please review this pr.. This pr will make the development more easier once this is merge I will send another pr for vscode and editor config.

**Comment by raajnadar** [comment]
I suppressed all the warnings which we can fix in future updates most of they are due to the usage of any keyword in typings.

We are left with 2 errors which I tried to fix but have no idea how to fix it.

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "publish-ssr": "cd examples/ssr && now && now alias react-multi-carousel.now.sh",
     "prettier": "prettier --write {examples,src,stories,test}/**/*.{js,ts,tsx}",
     "lint-staged": "lint-staged",
-    "lint": "tslint --fix \"src/**/*.ts\" \"src/**/*.tsx\" \"src/**/*.js\"",
+    "lint": "eslint src/**/*{.js,.jsx,.ts,.tsx}",
```
Comment (YIZHUANG) [inline package.json]: Remove ".js,.jsx" as this will cause the lint to fail.

**Comment by YIZHUANG** [comment]
Thank you for the PR, other than the ""eslint src/**/*{.js,.jsx,.ts,.tsx}" doesn't run, otherwise looks good.

**Comment by raajnadar** [comment]
I went through the package.json and found that husky is not configured I will create a separate PR with husky pre-commit hook and all the necessary commands.

I think we should not make this PR more complicated..!

**Pull request merged**